packages/teleport-types/src/uidl.ts

```diff
@@ -26,14 +26,25 @@ export interface ContextsUIDL {
   items?: Record<string, ContextUIDLItem>
 }
 
-export interface ResourceItemUIDL extends Resource {
-  id: string
+export interface UIDLENVValue {
+  type: 'env'
+  content: string
+}
+
+export interface UIDLResourceItem {
   name: string
+  headers?: Record<string, UIDLStaticValue | UIDLENVValue>
+  path: {
+    baseUrl: UIDLStaticValue | UIDLENVValue
+    route: UIDLStaticValue
+  }
+  params?: Record<string, UIDLStaticValue>
+  method?: 'GET' | 'POST'
+  body?: Record<string, UIDLStaticValue>
```

Comment:
@ionutpasca @vladnicula let's discuss more on this, we can generate resource files separately.  And map those resource fetchers from `initialProps` and `getStaticProps` and re-use them. Works as a base for generating fetchers in all other frameworks too